python/cudf/cudf/core/index.py

```diff
@@ -495,6 +493,10 @@ def dtype(self):
         dtype = np.dtype(np.int64)
         return _maybe_convert_to_default_type(dtype)
 
+    @property
+    def _dtypes(self) -> Iterator:
```

Comment:
*nit*: If the dtypes property returned `Iterable`, rather than `Iterator`, then this implementation could just be:

```
return [(self.name, self.dtype)]
```